packages/ui-navigation/src/AppNav/theme.js

```diff
@@ -22,8 +22,18 @@
  * SOFTWARE.
  */
 
-export default function generator({ borders, colors, spacing, typography }) {
-  return {
+/**
+ * Generates the theme object for the component from the theme and provided additional information
+ * @param  {Object} theme The actual theme object.
+ * @param  {Object} themeOverride User provided overrides of the default theme mapping.
+ * @return {Object} The final theme object with the overrides and component variables
+ */
+const generateComponentTheme = (theme, themeOverride = {}) => {
+  const { colors, borders, spacing, typography, key: themeName } = theme
+
+  const themeSpecificStyles = {}
+
+  const componentVariables = {
```

Comment:
Missing optional syntax: eg. typography?.fontFamily

Reply:
Not agaaaaaain